pytorch_lightning/accelerators/gpu.py

```diff
@@ -37,12 +37,14 @@ def setup_environment(self) -> None:
                 If the selected device is not GPU.
         """
         super().setup_environment()
-        if "cuda" not in str(self.root_device):
-            raise MisconfigurationException(f"Device should be GPU, got {self.root_device} instead")
-        torch.cuda.set_device(self.root_device)
+        if "cuda" not in str(self.training_type_plugin.root_device):
+            raise MisconfigurationException(
+                f"Device should be GPU, got {self.training_type_plugin.root_device} instead"
+            )
+        torch.cuda.set_device(self.training_type_plugin.root_device)
 
     def setup(self, trainer: "pl.Trainer") -> None:
-        self.set_nvidia_flags(trainer.local_rank)
+        self.set_nvidia_flags(getattr(self.training_type_plugin, "local_rank", 0))
```

Comment:
this can keep using trainer.local_rank as long as we have the trainer reference.